### PR TITLE
Added additional galaxy[tags] checks

### LINKED ISSUE #4409: galaxy[tags] does not align between ansible-lint and galaxy-importer
##### Summary

The galaxy-importer library has some additional regex checks on the tags attribute that ansible-lint is missing.

In galaxy-importer you can see a regex is defined for tags where ansible-lint does not:
https://github.com/ansible/galaxy-importer/blob/master/galaxy_importer/constants.py#L47

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.9.0 using ansible-core:2.17.5 ansible-compat:24.9.1 ruamel-yaml:0.18.6 ruamel-yaml-clib:None
A new release of ansible-lint is available: 24.9.0 → 24.10.0 Upgrade by running: pip install --upgrade ansible-lint
```

- ansible installation method: conda
- ansible-lint installation method: conda

##### STEPS TO REPRODUCE

Create basic galaxy.yml file:
```yaml
name: test
namespace: test
description: test
repository: https://repo.com
authors:
  - me
readme: README.md
version: 1.0.0
tags:
  - application
  - tag1
  - tag2
  - tag-3
```

And run `ansible-lint galaxy.yml`

```console (paste below)
ansible-lint galaxy.yml
```

However when importing into galaxy there is a failure on `tag-3`

##### Desired Behavior

ansible-lint should fail on `tag-3`

##### Actual Behavior

ansible-lint passes validation

```paste below
Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
A new release of ansible-lint is available: 24.9.0 → 24.10.0 Upgrade by running: pip install --upgrade ansible-lint
```

### PR DESCRIPTION
**Pull request by minsis**
* Added tag pattern check from galaxy-importer as `galaxy[tags-format]`
* Added max tag count from galaxy-importer as `galaxy[tags-count]`
* Added max tag character count from galaxy-importer as `galaxy[tags-length]`

Closes #4409

### CODE REVIEW

**Review by ssbarnea — APPROVED** [review]

**Comment by audgirka** [comment]
code coverage upload issue reported [here](https://github.com/codecov/codecov-action/issues/1705)

**Comment by minsis** [comment]
> code coverage upload issue reported [here](https://github.com/codecov/codecov-action/issues/1705)

Is this blocking the merge of this PR?

File: src/ansiblelint/constants.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
 """Constants used by AnsibleLint."""
 
+import re
 from enum import Enum
 from pathlib import Path
 from typing import Literal
 
+TAG_NAME_REGEXP = re.compile(r"^(?!.*__)[a-z]+[0-9a-z_]*$")
```
Comment (Qalthos) [inline src/ansiblelint/constants.py:8]:
```suggestion
TAG_NAME_REGEXP = re.compile(r"^(?!.*__)[a-z][0-9a-z_]*$")
```

I think removing the `+` here should avoid the regex issues being reported, while matching the same strings.

**Review by Qalthos — DISMISSED** [review]

File: src/ansiblelint/constants.py (hunk 1 above)
Comment (minsis) [reply src/ansiblelint/constants.py:8]: What regex errors? This was an exact copy and paste from galaxy importer lib.

**Comment by minsis** [comment]
All checks are passing now.

**Review by audgirka — APPROVED** [review]

**Pull request merged**